contributingGuides/PERFORMANCE.md

```diff
@@ -1,52 +1,275 @@
+
+#### Establishing a Test Flow and Baseline
+
+Before diving into detailed profiling, it's essential to establish a consistent "test flow" or reproduction steps for the performance issue. This ensures that measurements are consistent and comparable. Additionally, capturing "baseline measurements" is crucial. This involves noting key meta-information about the testing environment and application state, such as:
+
+*   The specific user account and its state.
+*   The exact commit SHA of the codebase being tested.
+*   Network conditions (online/offline, throttling).
+*   The platform(s) and device types used (simulator/emulator vs. physical device).
+*   The type of build (development vs. optimized production build).
+
+Maintaining a stable setup for these baseline measurements is critical for accurate comparisons and effective performance optimization.
+
+- You can test performance in development mode, but keep in mind it’s not optimized — things like loggers, debug tools, and extra checks can slow things down or cause jank that won’t happen in production. This can skew results, meaning you might misinterpret what’s slow in your app. For accurate metrics, always verify with a production build.
+- Use [`React.memo`](https://react.dev/reference/react/memo), [`useMemo`](https://react.dev/reference/react/useMemo), and [`useCallback`](https://react.dev/reference/react/useCallback) to prevent expensive re-renders.
 - Using a combination of [React DevTools Profiler](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) and [Chrome Dev Tools Performance Timing](https://calibreapp.com/blog/react-performance-profiling-optimization) can help identify unnecessary re-renders. Both tools can be used to time an interaction like the app starting up or navigating to a new screen.
 - Watch out for [very large lists](https://reactnative.dev/docs/optimizing-flatlist-configuration) and things like `Image` components re-fetching images on render when a remote uri did not change.
-- Avoid the temptation to over-optimize. There is added cost in both code complexity and performance when adding checks like `shouldComponentUpdate()`. Be selective about when you use this and make sure there is a measurable difference before proposing the change. As a very general rule, it should be measurably faster to run logic to avoid the re-render (e.g. do a deep comparison) than it would be to let React take care of it without any extra intervention from us.
+- Avoid the temptation to over-optimize. There is added cost in both code complexity and performance when using memoization hooks like `useMemo` and `useCallback`. Be selective about when you use them and make sure there is a measurable difference before proposing the change. As a very general rule, it should be measurably faster to run the memoized logic than it would be to let React re-render the component without any extra intervention from us.
```

Comment:
Tough one, but could we give a clearer guideline for deciding whether to use a memo, making it more black and white for contributors?